compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_cast_gen.bal

```diff
@@ -0,0 +1,149 @@
+        mv.visitTypeInsn(CHECKCAST, STRING_VALUE);
+    } else {
+        error err = error(io:sprintf("Casting is not supported from '%s' to 'string'", sourceType));
+        panic err;
+    }
+}
+
+function generateCastToBoolean(jvm:MethodVisitor mv, bir:BType sourceType) {
+    if (sourceType is bir:BTypeAny || sourceType is bir:BTypeAnyData) {
+        mv.visitTypeInsn(CHECKCAST, BOOLEAN_VALUE);
+        mv.visitMethodInsn(INVOKEVIRTUAL, BOOLEAN_VALUE, "booleanValue", "()Z", false);
+    } else {
+        error err = error(io:sprintf("Casting is not supported from '%s' to 'boolean'", sourceType));
+        panic err;
+    }
+}
+
+function generateCastToByte(jvm:MethodVisitor mv, bir:BType sourceType) {
+    if (sourceType is bir:BTypeAny || sourceType is bir:BTypeAnyData) {
+        mv.visitTypeInsn(CHECKCAST, BYTE_VALUE);
+        mv.visitMethodInsn(INVOKEVIRTUAL, BYTE_VALUE, "byteValue", "()B", false);
+    } else {
+        error err = error(io:sprintf("Casting is not supported from '%s' to 'byte'", sourceType));
+        panic err;
+    }
+}
+
+function generateCastToAny(jvm:MethodVisitor mv, bir:BType sourceType) {
+    if (sourceType is bir:BTypeInt) {
+        mv.visitMethodInsn(INVOKESTATIC, LONG_VALUE, "valueOf", io:sprintf("(J)L%s;", LONG_VALUE), false);
+    } else if (sourceType is bir:BTypeFloat) {
+        mv.visitMethodInsn(INVOKESTATIC, DOUBLE_VALUE, "valueOf", io:sprintf("(D)L%s;", DOUBLE_VALUE), false);
+    } else if (sourceType is bir:BTypeBoolean) {
+        mv.visitMethodInsn(INVOKESTATIC, BOOLEAN_VALUE, "valueOf", io:sprintf("(Z)L%s;", BOOLEAN_VALUE), false);
+    } else if (sourceType is bir:BTypeByte) {
+        mv.visitMethodInsn(INVOKESTATIC, BYTE_VALUE, "valueOf", io:sprintf("(B)L%s;", BYTE_VALUE), false);
+    } else {
+        // do nothing
+        return;
+    }
+}
+
+function generateCastToAnyData(jvm:MethodVisitor mv, bir:BType sourceType) {
+    if (sourceType is bir:BTypeAny) {
+        checkCast(mv, sourceType, "anydata");
+    } else {
+        // if value types, then ad box instruction
+        generateCastToAny(mv, sourceType);
+    }
+}
+
+function generateCastToUnionType(jvm:MethodVisitor mv, bir:BType sourceType, bir:BUnionType targetType) {
+    if (sourceType is bir:BTypeAny || sourceType is bir:BTypeAny) {
+        // TODO: add JSON
+        checkCast(mv, sourceType, targetType);
+    } else {
+        // if value types, then ad box instruction
+        generateCastToAny(mv, sourceType);
+    }
+}
+
+function generateCheckCast(jvm:MethodVisitor mv, bir:BType sourceType, bir:BType targetType, string targetClass) {
+    if (sourceType is bir:BTypeAny || sourceType is bir:BTypeAnyData) {
+        mv.visitTypeInsn(CHECKCAST, targetClass);
+    } else {
+        checkCast(mv, sourceType, targetType);
+        return;
+    }
+}
+
+function checkCast(jvm:MethodVisitor mv, bir:BType sourceType, bir:BType targetType) {
+    // TODO: invoke checkIsType() native function
```

Comment:
shall we call the static function that does the cast at runtime in JVM here directly ?